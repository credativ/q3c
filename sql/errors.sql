SELECT * FROM test_small WHERE q3c_poly_query(ra, dec,
'((1.000000,0.000000),
(0.998210,0.059804),
(0.992847,0.119394),
(0.983930,0.178557),
(0.971490,0.237080),
(0.955573,0.294755),
(0.936235,0.351375),
(0.913545,0.406737),
(0.887586,0.460642),
(0.858449,0.512899),
(0.826239,0.563320),
(0.791071,0.611724),
(0.753071,0.657939),
(0.712376,0.701798),
(0.669131,0.743145),
(0.623490,0.781831),
(0.575617,0.817719),
(0.525684,0.850680),
(0.473869,0.880596),
(0.420357,0.907359),
(0.365341,0.930874),
(0.309017,0.951057),
(0.251587,0.967835),
(0.193256,0.981148),
(0.134233,0.990950),
(0.074730,0.997204),
(0.014959,0.999888),
(-0.044865,0.998993),
(-0.104528,0.994522),
(-0.163818,0.986491),
(-0.222521,0.974928),
(-0.280427,0.959875),
(-0.337330,0.941386),
(-0.393025,0.919528),
(-0.447313,0.894377),
(-0.500000,0.866025),
(-0.550897,0.834573),
(-0.599822,0.800134),
(-0.646600,0.762830),
(-0.691063,0.722795),
(-0.733052,0.680173),
(-0.772417,0.635116),
(-0.809017,0.587785),
(-0.842721,0.538351),
(-0.873408,0.486989),
(-0.900969,0.433884),
(-0.925304,0.379225),
(-0.946327,0.323210),
(-0.963963,0.266037),
(-0.978148,0.207912),
(-0.988831,0.149042),
(-0.995974,0.089639),
(-0.999552,0.029915),
(-0.999552,-0.029915),
(-0.995974,-0.089639),
(-0.988831,-0.149042),
(-0.978148,-0.207912),
(-0.963963,-0.266037),
(-0.946327,-0.323210),
(-0.925304,-0.379225),
(-0.900969,-0.433884),
(-0.873408,-0.486989),
(-0.842721,-0.538351),
(-0.809017,-0.587785),
(-0.772417,-0.635116),
(-0.733052,-0.680173),
(-0.691063,-0.722795),
(-0.646600,-0.762830),
(-0.599822,-0.800134),
(-0.550897,-0.834573),
(-0.500000,-0.866025),
(-0.447313,-0.894377),
(-0.393025,-0.919528),
(-0.337330,-0.941386),
(-0.280427,-0.959875),
(-0.222521,-0.974928),
(-0.163818,-0.986491),
(-0.104528,-0.994522),
(-0.044865,-0.998993),
(0.014959,-0.999888),
(0.074730,-0.997204),
(0.134233,-0.990950),
(0.193256,-0.981148),
(0.251587,-0.967835),
(0.309017,-0.951057),
(0.365341,-0.930874),
(0.420357,-0.907359),
(0.473869,-0.880596),
(0.525684,-0.850680),
(0.575617,-0.817719),
(0.623490,-0.781831),
(0.669131,-0.743145),
(0.712376,-0.701798),
(0.753071,-0.657939),
(0.791071,-0.611724),
(0.826239,-0.563320),
(0.858449,-0.512899),
(0.887586,-0.460642),
(0.913545,-0.406737),
(0.936235,-0.351375),
(0.955573,-0.294755),
(0.971490,-0.237080),
(0.983930,-0.178557),
(0.992847,-0.119394),
(0.998210,-0.059804))'::polygon);
SELECT * FROM test_small WHERE q3c_poly_query(ra, dec, ARRAY[1,2,3,4,5,6,7]);
SELECT * FROM test_small WHERE q3c_poly_query(ra, dec, ARRAY[1,2,3,4]);
SELECT * FROM test_small WHERE q3c_poly_query(ra, dec, '((1,2),(3,4))'::polygon);
SELECT * FROM test_small WHERE q3c_poly_query(ra, dec, '((0,0),(90,0),(45,89))'::polygon);

