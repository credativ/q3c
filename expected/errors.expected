ERROR:  Polygons with more than 100 vertices are not supported
CONTEXT:  SQL function "q3c_poly_query" during inlining
ERROR:  Invalid array argument!
The array should contain even number of elements
CONTEXT:  SQL function "q3c_poly_query" during inlining
ERROR:  Invalid polygon! The polygon must have more than two vertices
CONTEXT:  SQL function "q3c_poly_query" during inlining
